starter-code/src/App.js

```diff
@@ -1,19 +1,74 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import Header from './components/misc/Header'
+import ProductBox from './components/ProductBox';
+
+import foods from './datasets/foods.json';
+import SearchBar from './components/SearchBar';
+import Menu from './components/Menu';
+import Form from './components/Form';
+
 
 class App extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      foods:[...foods],
+      menu: [],
+      addFood: false
+    }
+  }
+
+  onFilter = (search) => {
+    const newFoods = foods.filter(food => food.name.toLowerCase().includes(search.toLowerCase()));
+    this.setState({ foods: newFoods});
+  }
+
+  onModifyMenu = (food) => {
+    //quitamos la comida que vamos a meter para no meterla dos veces
+    const oldMenu = this.state.menu.filter(f=> {
+      return f.name != food.name
+    });
+    //nos creamos un  nuevo array con el anterior mas la nueva comida
+    const newMenu = [...oldMenu, food];
+    this.setState({
+      menu:newMenu
```

Comment:
Por qué no haces el `spread operator` directamente aquí? `this.setState({ menu: [...oldMenu, food] })`